### DISCUSSION TITLE
Testing in AdonisJS5

### DISCUSSION THREAD

**Discussion by AlexandreGerault** [adonisjs/core#1921] [Help] [answered]
Hi!

I just wonder, after reading [this article](https://preview.adonisjs.com/blog/running-tests-in-adonisjs-v5) and [this doc](https://github.com/thetutlage/japa) if there are/will be methods in Adonis to perform assertions just like we do with Laravel (`assertDatabaseHas('table', [values])`).

If not, is there any resource I'm missing to learn how to extend the tests tools to add new assertion methods?

Thanks

**Comment by thetutlage** [comment, marked as answer]
Hello @AlexandreGerault 

We use [chai.js assertion module](https://www.chaijs.com/api/assert) for assertions. They have an extendible API that you can use to add custom methods to it. I recommend reading their docs.

Regarding `assertDatabaseHas('table', [values])`. You basically have to add a new assertion method that uses the `Database` module of AdonisJS to make the query. Since, db queries are async, the assertion has to be called as `await assert.databaseHas()`.

**Reply by AlexandreGerault** [reply]
I'll try to write some useful assertions. Hope it can be hopeful someday :smile:
